swig/python2/examples/rpc_example.py

```diff
@@ -62,15 +62,14 @@ def print_value(value):
     else:
         print "(unprintable)"
 
-def test_rpc_cb(xpath, input_vals, input_cnt, output_vals, output_cnt, private_ctx):
+def test_rpc_cb(xpath, in_vals, out_vals, private_ctx):
 #    try:
         print "\n\n ========== RPC CALLED =========="
-        in_vals = sr.Vals(input_vals, input_cnt)
-        out_vals = sr.Vals(output_vals, output_cnt, 3)
-
         if (in_vals == None or out_vals == None):
             return
 
+        out_vals.allocate(3)
+
         for n in range(in_vals.val_cnt()):
             print_value(in_vals.val(n))
 
```

Comment:
I think that for Python a more suitable solution would be to return the output using the return statement and to avoid using output-like arguments. So something like:

def test_rpc_cb(xpath, in_vals, private_ctx)"
    out_vals = Vals(3)
    out_vals.val(0).set("/ietf-interfaces:interfaces/interface[name='eth0']", SR_LIST_T)
    out_vals.val(1).set("/ietf-interfaces:interfaces/interface[name='eth0']", SR_LIST_T)
    out_vals.val(2).set("/ietf-interfaces:interfaces/interface[name='eth0']", SR_LIST_T)

    return out_vals

Would it be possible to achieve it using C++ & SWIG?